### PR TITLE
Artifactory Pull Latest Compatible Version

### PR DESCRIPTION
**Pull request by sshookman**
[#110]

### Solution
By specifying "LATEST" as the version number, skelebot artifactory component will search for the latest compatible version of an artifact and select it for download.

This PR also add the ability to override the existing artifact file with the override parameter (`-o --override`).

It also fixes up some ugly error messaging.

### CODE REVIEW

**Comment by sshookman** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- skelebot/objects/semver.py  6
- skelebot/components/artifactory.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/skelebot/skelebot/pullRequest?prid=4510029&bid=15197710)

File: skelebot/components/artifactory.py [hunk 1]
```diff
@@ -32,18 +33,43 @@ def pushArtifact(artifactFile, user, token, file, url, force):
...
-def pullArtifact(user, token, file, url):
+def pullArtifact(user, token, file, url, override, original):
     """Pulls the given file from the url with the provided user/token auth"""
 
     if (artifactory.ArtifactoryPath(url, auth=(user, token)).exists()):
         print("Pulling {file} from {url}".format(file=file, url=url))
         path = artifactory.ArtifactoryPath(url, auth=(user, token))
         with path.open() as fd:
-            with open(file, "wb") as out:
+            dest = original if (override) else file
```
Comment (jagmoreira) [inline skelebot/components/artifactory.py]: What happens if the user specifies `override` but there's no artifact in the config file, or the artifact path doesn't exist yet? Can that even happen, i.e., can `original` be None?

Comment (sshookman) [reply skelebot/components/artifactory.py]: Also, there must be an artifact in the config file. They won't be able to even issue this command without it.

Comment (jagmoreira) [reply skelebot/components/artifactory.py]: Ah, so `original` will never be None. That makes sense.

**Pull request merged**